File: Completed/JavaScript/589.js
/**
 * // Definition for a Node.
 * function Node(val,children) {
 *    this.val = val;
 *    this.children = children;
 * };
 */
/**
 * @param {Node} root
 * @return {number[]}
 */
var preorder = function(root, arr = []) {
    if (root === null) return arr
    arr.push(root.val)
    root.children.forEach(c => {
        preorder(c, arr)
    })
    return arr
};

File: Completed/Go/589.go
/**
 * Definition for a Node.
 * type Node struct {
 *     Val int
 *     Children []*Node
 * }
 */

 func preorder(root *Node) []int {
    trav := []int{}
    helper(root, &trav)
    
    return trav
}

func helper(root *Node, trav *[]int) {
    if root == nil {
        return
    }
    
    *trav = append(*trav, root.Val)
    for _, child := range root.Children {
        helper(child, trav)
    }
}